src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -165,3 +166,140 @@ def do_observation(self, event) -> None:
         pop["census_year"] = event.time.year
 
         self.responses = pd.concat([self.responses, pop])
+
+
+class WICObserver(BaseObserver):
+    """Class for observing columns relevant to WIC administrative data.
+    """
+
+    def __repr__(self):
+        return f"WICObserver()"
+
+    @property
+    def name(self):
+        return f"wic_observer"
+
+    @property
+    def output_filename(self):
+        return f"wic.hdf"
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self.time_step = builder.configuration.time.step_size  # in days
+        assert 1 <= self.time_step <= 30, 'WICObserver requires model specification configuration with 1 <= time.step_size <= 30'
+        self.randomness = builder.randomness.get_stream(self.name)
+
+    def get_population_view(self, builder) -> PopulationView:
+        """Get the population view to be used for observations"""
+        return builder.population.get_view(columns=metadata.WIC_OBSERVER_COLUMNS_USED)
```

Comment:
@rmudambi I like the idea of moving these out of metadata and just keeping them defined as part of the class. thoughts? (That's what I did for household surveys)